CMakeLists.txt

```diff
@@ -97,20 +97,9 @@ else()
   message(STATUS "Using eigen_catkin's Eigen in ${EIGEN_CATKIN_EIGEN_INCLUDE}")
 endif()
 
-cs_add_library(${PROJECT_NAME} src/dependency_tracker.cc)
-
-if(NOT USE_SYSTEM_EIGEN)
-  add_dependencies(${PROJECT_NAME} eigen_src)
-endif()
-
-cs_install()
-
 if(NOT USE_SYSTEM_EIGEN)
   install(DIRECTORY ${EIGEN_CATKIN_EIGEN_INCLUDE}
     DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}
```

Comment:
Oh my, this is going to be the major pain here IMO. The installed `eigen_catkinConfig.cmake` will not contain this include dir, `${CATKIN_GLOBAL_INCLUDE_DESTINATION}`. Currently I only see on option: fix it in a cmake extra file and have two different (configured) versions: one for the devel space and one for the install case. It could be that we can find some cmake magic that can do it within the `eigen_catkinConfig.cmake` but it is probably tricky. Maybe a generator expression that is sensitive to the different situations. But how to distinguish the context?

In any case we should test the final .deb packages with a hidden workspace (e.g. moved away) or on a separate system (for test [this](https://github.com/ethz-asl/eigen_catkin_tools) could be a nice choice).

Reply:
You're right, it currently doesn't work in both cases. I tested it in a VM with your test package. Using the system eigen variant is fine and gives the following output:
```
eigen_catkin is using Eigen in '/usr/include/eigen3'.
The version there is 3.2.92.
```

However, creating a package with a downloaded eigen version will make your test package fail to compile as the eigen include dir isn't set correctly.